File: backend/src/main/java/com/coffeecommits/brakket/sponsorship/service/PanelComercialService.java
package com.coffeecommits.brakket.sponsorship.service;

import com.coffeecommits.brakket.sponsorship.dto.MetricasPatrocinioResponse;
import com.coffeecommits.brakket.sponsorship.dto.PanelComercialResponse;
import org.springframework.security.core.Authentication;

public interface PanelComercialService {

    PanelComercialResponse obtenerResumen(Authentication authentication);

    MetricasPatrocinioResponse obtenerMetricas(Authentication authentication, Long patrocinioId);
}

File: frontend/src/app/features/sponsorships/services/panel-comercial.service.ts
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

import { ApiService } from '../../../core/services/api.service';
import { MetricasPatrocinio, PanelComercial } from '../../../models/panel-comercial.model';

@Injectable({ providedIn: 'root' })
export class PanelComercialService {
  private readonly api = inject(ApiService);

  obtenerResumen(): Observable<PanelComercial> {
    return this.api.get<PanelComercial>('/sponsors/me/panel');
  }

  obtenerMetricas(patrocinioId: number): Observable<MetricasPatrocinio> {
    return this.api.get<MetricasPatrocinio>(`/sponsors/me/panel/metricas?patrocinioId=${patrocinioId}`);
  }
}
